flat_mutation_reader.hh

```diff
@@ -143,11 +143,12 @@ public:
 
                 if constexpr (std::is_same_v<future<stop_iteration>, decltype(consumer(pop_mutation_fragment()))>) {
                     return consumer(pop_mutation_fragment());
+                } else {
+                    auto result = stop_iteration::no;
+                    while ((result = consumer(pop_mutation_fragment())) != stop_iteration::yes && !is_buffer_empty() && !need_preempt())
+                        ;
```

Comment:
style nitpick: I think that we have a strict policy of always putting curly braces around bodies of loops and if statements

Reply:
I thought that this way is more readable than while() {} but will change.

Reply:
Did you measure any improvement attributed to this tight loop vs. simply returning a `make_ready_future<stop_iteration>(consumer(pop_mutation_fragment()))`?

The repeat loop checks the result and need_preempt anyhow and then the next iteration of the inner loop will check for `is_buffer_empty`.